mapper/orm/src/main/java/org/hibernate/search/mapper/orm/event/impl/HibernateSearchEventListener.java

```diff
@@ -128,39 +141,52 @@ public void onPostUpdate(PostUpdateEvent event) {
 			return;
 		}
 
+		boolean considerAllDirty;
 		BitSet dirtyPaths;
+		BitSet dirtyDirectAssociationPaths;
 		if ( dirtyCheckingEnabled ) {
 			int[] dirtyProperties = event.getDirtyProperties();
 			if ( dirtyProperties == null || dirtyProperties.length == 0 ) {
```

Comment:
It might, but that's also an unlikely scenario. *Absolutely* all properties being changed, all at once, in the same transaction... I can't see that being frequent save for a few exotic applications.
Feel free to create a ticket if you think there's something to be explored, though.